src/AspNet.Security.OAuth.Weixin/WeixinAuthenticationHandler.cs

```diff
@@ -79,6 +79,16 @@ protected override async Task<HandleRequestResult> HandleRemoteAuthenticateAsync
             throw new HttpRequestException("An error occurred while retrieving user information.");
         }
 
+        (var openId, var unionId) = GetUserIdentifier(payload.RootElement);
+        if (!string.IsNullOrWhiteSpace(unionId))
+        {
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, unionId, ClaimValueTypes.String, Options.ClaimsIssuer));
+        }
+        else
+        {
+            identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, openId!, ClaimValueTypes.String, Options.ClaimsIssuer));
+        }
```

Comment:
I think something like this would be a bit more DRY:

```suggestion
        (var openId, var unionId) = GetUserIdentifier(payload.RootElement);
        var nameIdentifier = !string.IsNullOrWhiteSpace(unionId) ? unionId : openId;
        identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, nameIdentifier!, ClaimValueTypes.String, Options.ClaimsIssuer));
```

Reply:
Yes, this is great, I'm very busy right now, I try to take the time to provide a test case